api/virtual_accounts/virtual_accounts.go

```diff
@@ -52,40 +63,48 @@ func NewVAHandler(jwt *tokens.JWT, db *sql.DB) *VAHandler {
 	return &VAHandler{jwt, db}
 }
 
+// Delete VAC and checkVaNumValid made by Joseph
+// Delete VAC relies heavily on query logic so unit testing is not really the perfect choice
+// unless we create factory and interface that enables memory storage
+func CheckVaNumValid(vaNum string) bool {
+	if len(vaNum) == 13 {
+		return true
+	}
+	return false
+}
+
 func (vh VAHandler) DeleteVac(w http.ResponseWriter, r *http.Request) {
 	token := vh.jwt.GetToken(r)
 	err := token.Valid()
 	if err != nil {
-		helpers.HTTPError(w, http.StatusBadRequest, err.Error())
+		helpers.HTTPError(w, http.StatusBadRequest, TokenExpires)
 	}
 
-	var reqBody DeleteVacRequest
-	err = json.NewDecoder(r.Body).Decode(&reqBody)
-	if err != nil {
-		helpers.HTTPError(w, http.StatusBadRequest, "Unable to decode request body")
+	vaNum := chi.URLParam(r, "va_num")
+	if !CheckVaNumValid(vaNum) {
+		helpers.HTTPError(w, http.StatusBadRequest, InvalidVaNumber)
 		return
 	}
 
-	cust, err := database.GetCustomerById(vh.db, token.CustId)
+	trx, err := vh.db.Begin()
 	if err != nil {
-		helpers.HTTPError(w, http.StatusBadRequest, "User not found")
-		return
+		helpers.HTTPError(w, http.StatusInternalServerError, FailSqlTransaction)
 	}
+	defer trx.Rollback()
 
-	vac, err := database.GetVacByAccountNum(vh.db, cust.AccountNum)
+	vac, err := database.GetVacByAccountNum(vh.db, token.AccountNum)
+	if err != nil {
+		helpers.HTTPError(w, http.StatusNotFound, VANotFound)
+		return
+	}
 
+	err = database.RevertVacBalanceToMainAccount(vh.db, vac)
```

Comment:
If you want to use trx, pass trx in place of db as a parameter in this function. Otherwise, the query will not be run by trx, but by vh.db instead thus your query won't be put together inside one transaction.